pkg/operator/controller/externaldns/deployment.go

```diff
@@ -198,6 +201,14 @@ func desiredExternalDNSDeployment(namespace, image, secretName string, serviceAc
 		depl.Spec.Template.Spec.Containers = []corev1.Container{*container}
 	} else {
```

Comment:
ZoneID is an optional field of the API. So we are not guaranteed to have it in the CR which disables the guess of the Azure provider based on the resource path pattern.

CC @Miciah

Reply:
If an empty zones list means "publish in all zones", then we need two external-dns instances on Azure when the zones list is empty: one for public zones and one for private zones.  

We should also make sure that the godoc in the API is clear about the semantics of an empty zones list, namely that the operator configures external-dns to publish to all zones if the zones list is empty unless the operator is running on a platform on which the operator can look up a default set of zones, viz. OpenShift with its cluster DNS config.

Reply:
After the discussion with everybody: 

- If the zones list is empty, create one external-dns container for public and one for private.
- Else, create one external-dns container per zone, configured as public or private in accordance with the zone.